spark/dl/src/main/scala/com/intel/analytics/bigdl/models/utils/ModelBroadcast.scala

```diff
@@ -89,12 +90,17 @@ class ModelBroadcast[T: ClassTag](implicit ev: TensorNumeric[T]) extends Seriali
   }
 
   private def putWeightBias(broadcastWeightBias: Array[Tensor[T]],
-    localModel: Module[T]): Unit = {
-    val localWeightBias = localModel.parameters()._1
+    localModel: Module[T], inference: Boolean = true): Unit = {
+    val (localWeightBias, localGradWeightBias) = localModel.parameters()
     var i = 0
     while (i < localWeightBias.length) {
       if (localWeightBias(i) != null) {
-        localWeightBias(i).set(broadcastWeightBias(i))
+        if (!inference) {
+          localWeightBias(i).resizeAs(broadcastWeightBias(i)).copy(broadcastWeightBias(i))
```

Comment:
Even in training, we can still share weight and bias, no?

Reply:
Seems good, and this can reduce some time for `putWeights`. Maybe also increase cache hits ratio.
Let's do it later in #1634.

Reply:
I just check the code of initThreadModels.
The weight is shared after all sub models are cloned.
```
      val weights = cached.head._2
      cached.map(c =>
        if (!c._2.eq(weights)) {
          c._2.storage().set(weights.storage())
        }
      )
```